src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java

```diff
@@ -26,173 +27,209 @@
 	}
 
 	@Test
-	void sendIncident() throws JsonProcessingException {
-		final var incidentSaveRequest = buildIncidentSaveRequest(Category.LIVBOJ);
-		final var incidentEntity = buildIncidentEntity(Category.LIVBOJ);
-		when(incidentRepository.save(any())).thenReturn(incidentEntity);
-		when(lifeBuoyIntegration.sendLifeBuoy(any(IncidentDto.class))).thenReturn("SP_20220826_30694");
-
-		final var response = incidentService.sendIncident(incidentSaveRequest);
-		assertThat(response).isNotNull();
-		assertThat(response.getStatusText()).isEqualTo(Status.INSKICKAT.getLabel());
-		verify(incidentRepository, times(1)).save(any());
-		verify(lifeBuoyIntegration, times(1)).sendLifeBuoy(any(IncidentDto.class));
+	void fetchOepIncidentStatusTest() {
+		when(mockIncidentRepository.findIncidentEntityByExternalCaseId(anyString()))
+			.thenReturn(Optional.ofNullable(createIncidentEntity()));
+
+		var result = incidentService.fetchOepIncidentStatus(anyString());
 
+		assertThat(result).isNotNull().isInstanceOf(IncidentOepResponse.class);
+		verify(mockIncidentRepository, times(1)).findIncidentEntityByExternalCaseId(any());
+		verifyNoMoreInteractions(mockIncidentRepository);
 	}
 
 	@Test
-	void sendMSVIncident() {
-		final var incidentSaveRequest = buildIncidentSaveRequest(Category.VATTENMATARE);
-		final var incidentEntity = buildIncidentEntity(Category.VATTENMATARE);
-		when(incidentRepository.save(any())).thenReturn(incidentEntity);
-
-		doNothing().when(messagingIntegration).sendMSVAEmail(any(IncidentDto.class));
-
-		final var response = incidentService.sendIncident(incidentSaveRequest);
-		assertThat(response).isNotNull();
-		assertThat(response.getStatusText()).isEqualTo(Status.INSKICKAT.getLabel());
-		verify(incidentRepository, times(1)).save(any());
-		verify(messagingIntegration, times(1)).sendMSVAEmail(any(IncidentDto.class));
+	void fetchOepIncidentStatus_NotFoundTest() {
+		when(mockIncidentRepository.findIncidentEntityByExternalCaseId(anyString())).thenReturn(Optional.empty());
+
+		assertThatThrownBy(() -> incidentService.fetchOepIncidentStatus(anyString()))
+			.hasMessageContaining("Not Found: Incident with id:  not found")
+			.isInstanceOf(Problem.class);
 	}
 
 	@Test
-	void sendIncidentAndExceptionThrown() {
-		final var incidentSaveRequest = buildIncidentSaveRequest(Category.KLOTTER);
-		final var incidentEntity = buildIncidentEntity(Category.KLOTTER);
-		when(incidentRepository.save(any())).thenReturn(incidentEntity);
-		doThrow(new NullPointerException()).when(messagingIntegration).sendEmail(any(IncidentDto.class));
-
-		final var response = incidentService.sendIncident(incidentSaveRequest);
-		assertThat(response).isNotNull();
-		assertThat(response.getStatusText()).isEqualTo(Status.ERROR.getLabel());
-		verify(incidentRepository, times(1)).save(any());
-		verify(messagingIntegration, times(1)).sendEmail(any(IncidentDto.class));
+	void fetchPaginatedIncidentsTest() {
```

Comment:
Add test with for Optional.empty() to verify default values for page and pageSize?